subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/AnnotationProcessorDeclarationSerializer.java

```diff
@@ -25,11 +25,13 @@
     @Override
     public AnnotationProcessorDeclaration read(Decoder decoder) throws EOFException, Exception {
         String name = decoder.readString();
-        return new AnnotationProcessorDeclaration(name);
+        IncrementalAnnotationProcessorType type = IncrementalAnnotationProcessorType.values()[decoder.readSmallInt()];
```

Comment:
I don't remember where this is stored: is this on a per-Gradle version directory? If yes, then it's fine. Otherwise, there's a risk that if the enum is changed, when you read back, you actually map to the wrong enum value.

Reply:
All file content caches are per version, so this is safe.